app/server.js

```diff
@@ -37,32 +69,53 @@ app.get("/analyzeData", async(req, res) => {
     }
 
     //TODO get logged in user id
-    const userID = "42";
+    // const userID = "42";
+    //let token = window.localStorage.getItem("token");
+    //'x-auth=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6InRlc3Ryb2JlcnQifQ.ILGdmHYwNoen9CwS1KIR9aDS_nNNNUr8EITUM9Flges'
+    let user = await getUser(req.headers.cookie.substring(7));
```

Comment:
You shouldn't use `String.substring` because it makes assumptions about the contents and ordering of the cookie that may not always be correct. I'd suggest just using the getUser function that Tyler already created for this purpose https://github.com/cs326-final-project/digital-mirror/blob/dev/app/api/users.js#L53